### PR TITLE
plan: fix a topn bug

### PR DESCRIPTION
**Pull request by hanfei1991**
If a function exists in order by lists, it shouldn't push down anything. But we pushed a single limit into table scan , which may cause a bug about correctness. This PR fixes it.
@coocood @shenli @zimulala @tiancaiamao @XuHuaiyu

### CODE REVIEW

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -899,16 +899,18 @@ func (p *Sort) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo,
...
 			selfProp.props = append(selfProp.props, &columnProp{col: col, desc: by.Desc})
 		} else {
 			selfProp.props = nil
+			valid = false
 			break
 		}
 	}
 	selfProp.sortKeyLen = len(selfProp.props)
-	if len(prop.props) == 0 && prop.limit != nil {
+	if valid && len(prop.props) == 0 && prop.limit != nil {
```
Comment (coocood) [inline plan/physical_plan_builder.go]: How about check `len(self.props) == 0`?

**Comment by ngaut** [comment]
LGTM

**Comment by hanfei1991** [comment]
@ngaut  test case has been addad in executor_test.go

**Comment by coocood** [comment]
LGTM

**Pull request merged**